### PR TITLE
parser: turn some functions into member functions of model.Job

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: ref #31716

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: parser/model/ddl.go [hunk 1]
```diff
@@ -511,6 +511,56 @@ func (job *Job) NotStarted() bool {
...
+			return ok && needReorg
+		}
+		return false
+	default:
+		return false
+	}
+}
+
+// IsJobRollbackable checks whether the job can be rollback.
+func (job *Job) IsJobRollbackable() bool {
```
Comment (tangenta) [inline parser/model/ddl.go:531]:
```suggestion
func (job *Job) IsRollbackable() bool {
```

Comment (hawkingrei) [reply parser/model/ddl.go]: Done

**Review by tangenta — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]
LGTM. But it seems we still need one more reviewer of @pingcap/tidb-configuration-reviewer

**Pull request merged**